smriprep/workflows/anatomical.py

```diff
@@ -27,17 +28,194 @@
+
+
+      1. Creates brain mask assuming all zero voxels are outside the brain
+      2. Applies N4 bias field correction
+      3. (Optional) apply ATROPOS and massage its outputs
+      4. Use results from 3 to refine N4 bias field correction
+
+
+    .. workflow::
+        :graph2use: orig
+        :simple_form: yes
+
+        from smriprep.workflows.anatomical import init_n4_only_wf
+        wf = init_n4_only_wf()
+
+
+    **Parameters**
+
+        omp_nthreads : int
+            Maximum number of threads an individual process may use
+        mem_gb : float
+            Estimated peak memory consumption of the most hungry nodes
+        bids_suffix : str
+            Sequence type of the first input image. For a list of acceptable values
+            see https://bids-specification.readthedocs.io/en/latest/\
+04-modality-specific-files/01-magnetic-resonance-imaging-data.html#anatomy-imaging-data
+        atropos_refine : bool
+            Enables or disables the whole ATROPOS sub-workflow
+        atropos_use_random_seed : bool
+            Whether ATROPOS should generate a random seed based on the
+            system's clock
+        atropos_model : tuple or None
+            Allows to specify a particular segmentation model, overwriting
+            the defaults based on ``bids_suffix``
+        name : str, optional
+            Workflow name (default: antsBrainExtraction)
+
+
+    **Inputs**
+
+        in_files
+            List of input anatomical images to be bias corrected,
+            typically T1-weighted.
+            If a list of anatomical images is provided, subsequently
+            specified images are used during the segmentation process.
+            However, only the first image is used in the registration
+            of priors.
+            Our suggestion would be to specify the T1w as the first image.
+
+
+    **Outputs**
+
+
+        out_file
+            :abbr:`INU (intensity non-uniformity)`-corrected ``in_files``
+        out_mask
+            Calculated brain mask
+        bias_corrected
+            Same as "out_file", provided for consistency with brain extraction
+        bias_image
+            The :abbr:`INU (intensity non-uniformity)` field estimated for each
+            input in ``in_files``
+        out_segm
+            Output segmentation by ATROPOS
+        out_tpms
+            Output :abbr:`TPMs (tissue probability maps)` by ATROPOS
+
+
+    """
+    wf = pe.Workflow(name)
+
+    if omp_nthreads is None or omp_nthreads < 1:
+        omp_nthreads = cpu_count()
+
+    inputnode = pe.Node(niu.IdentityInterface(fields=['in_files', 'in_mask']),
+                        name='inputnode')
+
+    outputnode = pe.Node(niu.IdentityInterface(
+        fields=['out_file', 'out_mask', 'bias_corrected', 'bias_image',
+                'out_segm', 'out_tpms']),
+        name='outputnode')
+
+    # Create brain mask
+    thr_brainmask = pe.Node(
+        Binarize(thresh_low=10), name='binarize')
```

Comment:
This feels high, though I also don't have numbers here. For the image you show in https://github.com/poldracklab/smriprep/pull/116#issuecomment-541938810, what is the result of:

```Python
data = nb.load(fname).get_fdata()
np.max(data[data < 10])
```

If this is << 1 (this is my guess), I would much prefer that we use a small number such as 1 or even 0.1. That makes it clear that it's interpolation noise that we'll be masking out, not very dark CSF.

Reply:
I agree in principle, and I was hoping for some feedback as you have more expertise on what the range of values tend to be. 

But I think 0.1 is definitely too low, and 1 might still be. FWIW, at least on this image, all "real" values, even the darkest CSF were more like 100+.

I get:
```
In [6]: data = nb.load(fname).get_fdata()
   ...: np.max(data[data < 10])
   ...: 
Out[6]: 9.0
```

Reply:
If you try this with any value though you get a similar result. My guess is this is all happening near the edge of the brain, and most of the non-contiguous noise is maybe no higher than 5? It's hard to say from a single sample though. 

```
In [8]: data = nb.load(fname).get_fdata()
   ...: np.max(data[data < 20])
Out[8]: 19.0
```

Reply:
Perhaps we can get a better sense of this noise with a histogram?

```Python
import matplotlib.pyplot as plt
plt.hist(data[data < 20], bins=20)
```

IDK. I don't really have a sense of what the lowest CSF T1w values we're likely to find will be. OTOH, we can also just put this out and let people complain if we rip holes in their ventricles, since this is already a kind of an unrecommended hack.

> I was hoping for some feedback as you have more expertise on what the range of values tend to be.

Since we're not using quantitative MRI and getting actual T1 values, the scale is arbitrary. I suspect the range varies a lot across scanner. And any preprocessing that was done to get to the skull-stripped images is obviously free to rescale.

Reply:
![image](https://user-images.githubusercontent.com/2774448/67888271-d1421080-fb1a-11e9-81f2-f0580b0d10b6.png)
![image](https://user-images.githubusercontent.com/2774448/67888708-9c828900-fb1b-11e9-8c7d-be5a9d045493.png)

Okay looks like we could safely set the threshold much lower

Reply:
Maybe 1-2 is good.

Reply:
Yes. My thoughts on the check was that it could be done in fmriprep (or at least outside this wf) and in smriprep it could just be an off/on flag. That was the test can be outside of nipype and use pybids to quickly run it, and set the flag here. 

The test of checking in an individual image that all extreme corner values == 0 worked for me. It's just in the averaged image we are dealing with here that we start getting values slightly above 0.